oteljava/context-storage/src/main/scala/org/typelevel/otel4s/oteljava/context/IOLocalContextStorage.scala

```diff
@@ -75,34 +75,34 @@ object IOLocalContextStorage {
     * }
     *   }}}
     */
-  def localProvider[F[_]: LiftIO](implicit F: MonadCancelThrow[F]): LocalProvider[F, Context] =
+  def localProvider[F[_]: Console: LiftIO](implicit F: MonadCancelThrow[F]): LocalProvider[F, Context] =
     new LocalProvider[F, Context] {
       def local: F[Local[F, Context]] =
         ContextStorage.get() match {
           case storage: IOLocalContextStorage =>
             whenPropagationEnabled(storage.local)
 
-          case other if other.getClass.getName == AgentContextStorageClass =>
-            whenPropagationEnabled(agentLocal)
-
           case other =>
-            F.raiseError(
-              new IllegalStateException(
-                "IOLocalContextStorage is not configured for use as the ContextStorageProvider. " +
-                  s"The current storage: ${other.getClass.getName}."
-              )
-            )
+            getAgentLocalContext() match {
+              case Some(ioLocalJContext) =>
+                val ioLocal = ioLocalJContext.lens(ctx => Context.wrap(ctx))(_ => c => c.underlying)
+                val local = LocalProvider.localForIOLocal(ioLocal)
```

Comment:
I think there's a method on `IOLocal` for this now

```suggestion
                val local = ioLocal.asLocal
```

Reply:
`asLocal` returns `Local[IO, Context]`, while we need `Local[F, Context]` :(

Reply:
oh, my mistake